src/renderers/dom/client/ReactInputSelection.js

```diff
@@ -29,13 +37,10 @@ function isInDocument(node) {
  */
 var ReactInputSelection = {
 
-  hasSelectionCapabilities: function(elem) {
-    var nodeName = elem && elem.nodeName && elem.nodeName.toLowerCase();
-    return nodeName && (
-      (nodeName === 'input' && elem.type === 'text') ||
-      nodeName === 'textarea' ||
-      elem.contentEditable === 'true'
-    );
+  hasSelectionCapabilities: function(elem : HTMLElement) {
+    return (elem instanceof HTMLInputElement && elem.type === 'text') ||
```

Comment:
Why did you change this logic here? We shouldn't change implementation logic unless there's a good reason to do it. We also don't want to include changes like this alongside stylistic changes like flowtype annotations, as it would make it hard to revert just this change if we needed to.